setuptools/build_meta.py

```diff
@@ -117,11 +117,11 @@ def _file_with_extension(directory, extension):
     matching = (f for f in os.listdir(directory) if f.endswith(extension))
     try:
         (file,) = matching
-    except ValueError:
+    except ValueError as e:
         raise ValueError(
             'No distribution was found. Ensure that `setup.py` '
             'is not empty and that it calls `setup()`.'
-        )
+        ) from e
```

Comment:
I am a bit sceptical that chaining 2 `ValueError` would facilitate the debugging...
Probably the original idea here was to preserve the stack trace but avoid unnecessary verboseness.

Using `from None` would suppress the stack trace right? (or am I mistaken on how `raise ... from None` works?) Suppressing the stack trace would make it more difficult to debug...

Maybe we should just leave this one as it is...

Reply:
https://docs.python.org/3/tutorial/errors.html#exception-chaining

As far as I can understand, you get the 2 `ValueError` with or without chaining.  Just the wording differs. From a small example similar to the first one in the above lin:

```diff
 Traceback (most recent call last):
   File "/path/to/foobar.py", line 3, in <module>
     open("database.sqlite")
 FileNotFoundError: [Errno 2] No such file or directory: 'database.sqlite'
 
-During handling of the above exception, another exception occurred:
+The above exception was the direct cause of the following exception:
 
 Traceback (most recent call last):
   File "/path/to/foobar.py", line 5, in <module>
-    raise RuntimeError("unable to handle error")
+    raise RuntimeError("unable to handle error") from exc
 RuntimeError: unable to handle error
```

Reply:
Oh well, if that is the case, then the proposed changes should be fine, they don't change that much anyway.

Reply:
Thank you btw for clarifying my misunderstanding :)